ui/components/MeshModelRegistry/MesheryTreeView.js

```diff
@@ -43,6 +44,27 @@ const VersionedModelComponentTree = ({
     model: versionedModelDef.name,
     params: { version: versionedModelDef.model.version },
   });
+  const { selectedItemUUID } = useRegistryRouter();
+
+  useEffect(() => {
```

Comment:
An opportunity for use of RTK?

Reply:
This is not an RTK query, This is a custom hook that encapsulates the logic for managing routing and query parameters.